integration-tests/src/test/java/oracle/kubernetes/operator/MiiBaseTest.java

```diff
@@ -0,0 +1,119 @@
+        getWeblogicImageName() + ":" + getWeblogicImageTag());
+    domainMap.put("logHomeOnPV", "true");
+    //domainMap.put("wdtDomainType", "WLS");
+
+    if (prefix != null && !prefix.trim().equals("")) {
+      domainMap.put("image", prefix.toLowerCase() + "-modelinimage-" + suffixCount + ":latest");
+    } else {
+      domainMap.put("image", "modelinimage-" + suffixCount + ":latest");
+    }
+    return domainMap;
+  }
+
+  /**
+   * Create domain using model in image.
+   * @param domainUidPrefix domain UID prefix
+   * @param domainNS domain namespace
+   * @param wdtModelFile file should be under test/resouces/model-in-image dir,
+   *                     value can be ./model.wls.yaml
+   * @param wdtModelPropertiesFile file should be under test/resouces/model-in-image dir,
+   *                               value can be ./model.empty.properties
+   * @param cmFile creates configmap from this file or dir
+   */
+  public Domain createMiiDomainWithConfigMap(String domainUidPrefix,
+                                             String domainNS, String wdtModelFile, String wdtModelPropertiesFile,
+                                             String cmFile, String wdtDomainType) throws Exception {
+    Map<String, Object> domainMap =
+        createModelInImageMap(getNewSuffixCount(), domainUidPrefix);
+    // config map before deploying domain crd
+    String cmName = domainMap.get("domainUID") + "-mii-config-map";
+
+    domainMap.put("namespace", domainNS);
+    domainMap.put("wdtModelFile", wdtModelFile);
+    domainMap.put("wdtModelPropertiesFile", wdtModelPropertiesFile);
+    domainMap.put("wdtDomainType", wdtDomainType);
+
+    domainMap.put("miiConfigMap", cmName);
+    domainMap.put("miiConfigMapFileOrDir", cmFile);
+
+    Domain domain = TestUtils.createDomain(domainMap);
+    // domain = new Domain(domainMap, true, false);
+    domain.verifyDomainCreated();
+    return domain;
+  }
+
+  /**
+   * Modify the domain yaml to add reference to config map and change domain-level restart version.
+   * @param cmName Config map name
+   * @param domain the domain
+   * @throws Exception on failure
+   */
+  public void modifyDomainYamlWithNewConfigMapAndDomainRestartVersion(
+      String cmName, Domain domain)
+      throws Exception {
+    String originalYaml =
+        getUserProjectsDir()
+            + "/weblogic-domains/"
+            + domain.getDomainUid()
+            + "/domain.yaml";
+
+    // Modify the original domain yaml to include restartVersion in admin server node
+    DomainCrd crd = new DomainCrd(originalYaml);
+    Map<String, String> objectNode = new HashMap<>();
+    objectNode.put("restartVersion", "v1.1");
+    crd.addObjectNodeToDomain(objectNode);
+    String modYaml = crd.getYamlTree();
+    LoggerHelper.getLocal().log(Level.INFO, modYaml);
+
+    //change config map name to new config map
+    modYaml.replaceAll((String)domain.getDomainMap().get("miiConfigMap"), cmName);
```

Comment:
Return value of replaceAll is ignored.

Reply:
@jshum2479 ?

Reply:
@vanajamukkara  It should replace the modYaml

Reply:
It's not doing that presently since Java Strings are immutable.  The return value has the modified String